spring-kafka/src/test/java/org/springframework/kafka/kstreams/KStreamsTests.java

```diff
@@ -0,0 +1,182 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.processor.WallclockTimestampExtractor;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKStreams;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.annotation.KStreamDefaultConfiguration;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.test.rule.KafkaEmbedded;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.concurrent.SettableListenableFuture;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 2.0
+ */
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class KStreamsTests {
+
+	private static final String STREAMING_TOPIC1 = "streamingTopic1";
+
+	private static final String STREAMING_TOPIC2 = "streamingTopic2";
+
+	@ClassRule
+	public static KafkaEmbedded embeddedKafka = new KafkaEmbedded(1, true, 1, STREAMING_TOPIC1, STREAMING_TOPIC2);
+
+	@Autowired
+	private KafkaTemplate<Integer, String> kafkaTemplate;
+
+	@Autowired
+	private SettableListenableFuture<String> resultFuture;
+
+	@Test
+	public void testKStreams() throws Exception {
+		String payload = "foo" + UUID.randomUUID().toString();
+		String payload2 = "foo" + UUID.randomUUID().toString();
+
+		this.kafkaTemplate.sendDefault(0, payload);
+		this.kafkaTemplate.sendDefault(0, payload2);
+
+		String result = resultFuture.get(60, TimeUnit.SECONDS);
+
+		assertThat(result).isNotNull();
+
+		assertThat(result).isEqualTo(payload.toUpperCase() + payload2.toUpperCase());
+	}
+
+
+	@Configuration
+	@EnableKafka
+	@EnableKStreams
```

Comment:
Yes, there is a `@KafkaListener` configuration to poll message from the topic after KStream "magic".